google/cloud/internal/pagination_range.h

```diff
@@ -93,18 +93,17 @@ class PagedStreamReader {
    *   successful end of stream.
    */
   typename StreamReader<T>::result_type GetNext() {
-    if (current_ == page_.end()) {
-      if (last_page_) return Status{};
+    while (current_ == page_.end() && !last_page_) {
       request_.set_page_token(std::move(token_));
       auto response = loader_(request_);
       if (!response.ok()) return std::move(response).status();
       token_ = ExtractPageToken(*response);
       if (token_.empty()) last_page_ = true;
       page_ = extractor_(*std::move(response));
       current_ = page_.begin();
-      if (current_ == page_.end()) return Status{};
     }
-    return std::move(*current_++);
+    if (current_ != page_.end()) return std::move(*current_++);
+    return Status{};
```

Comment:
[Oops ... a comment seems to have been lost ... undoubtedly my mistake.]

Opt nit:  I would reverse the logic here.
```
    if (current_ == page_.end()) return Status{};
    return std::move(*current_++);
```

Reply:
Done.